src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Models/OpenIdApplication.cs

```diff
@@ -69,23 +78,26 @@ public class OpenIdApplication
         /// <summary>
         /// Gets or sets the callback URLs associated with the current application.
         /// </summary>
-        public ImmutableArray<string> RedirectUris { get; set; }
-            = ImmutableArray.Create<string>();
+        public ImmutableArray<string> RedirectUris { get; set; } = [];
 
         /// <summary>
         /// Gets or sets the requirements associated with the current application.
         /// </summary>
-        public ImmutableArray<string> Requirements { get; set; }
-            = ImmutableArray.Create<string>();
+        public ImmutableArray<string> Requirements { get; set; } = [];
 
         /// <summary>
         /// Gets or sets the roles associated with the application.
         /// </summary>
-        public ImmutableArray<string> Roles { get; set; }
-            = ImmutableArray.Create<string>();
+        public ImmutableArray<string> Roles { get; set; } = [];
 
         /// <summary>
-        /// Gets or sets the application type associated with the current application.
+        /// Gets or sets the settings associated with the application.
+        /// </summary>
+        public ImmutableDictionary<string, string> Settings { get; set; }
+            = ImmutableDictionary.Create<string, string>();
+
+        /// <summary>
+        /// Gets or sets the client type associated with the current application.
         /// </summary>
         public string Type { get; set; }
```

Comment:
Ideally, we should rename this property to `ClientType` but it's not clear to me what would be the proper way to do it with YesSql /cc @sebastienros

Reply:
And I think there is no need for two sessions/connections, if you get a connection after SaveChangesAsync()

Reply:
@kevinchalet, you can mark Type as Obsolete and add a new property called ClientType. Setting Type can also set ClientType if ClientType is null.

@MichaelPetrinolis you may want to use `await using` to dispose the connection asynchronously. Also, that could should be call that query using ShellScope.AddDeferredTask to ensure that query is called after everything has been setup (specially during new setup)

Reply:
@MikeAlhayek the Deferred Tasks execute after the migrations complete? If this is the case and the migration fails, subsequently the deferred task will also fail, as the DB won't have the new index table column name (if we also rename the column name 'Type' to 'ClientType' in the index). Maybe adding a DataMigration step that executes after schema migrations (considering the current schema version) would be handy.

Reply:
Yes differed task will be executed at the end of the request. You can call it after the alter command. If the alter exceptionally failed, the next line will not be called.

Reply:
@MikeAlhayek the issue with the deferred task is that the schema might get updated, but the data are not updated if an error during the update occurs. I think we need a two-step migration process, one to update the schema (existing) and one to run data migrations for the new Schema Version as deferred task.

Reply:
@MichaelPetrinolis @MikeAlhayek thanks for your suggestions!

That said, it looks a bit risky and it's likely I won't have too much time to troubleshot any potential issue that may occur due to this change.

If anyone wants to submit a PR to change the name of the `Type` column to `ClientType` (once OpenIddict 5.0 ships next week), please feel free! 😃